blog/2023-04-23-json.md

```diff
@@ -0,0 +1,172 @@
+
+Wing has been tackling the complexities of distributed computing in the cloud from various angles. One of the most notable angles is the concept of inflight functions, which serve as Wing's distributed computing primitives. These are code blocks packaged up and executed on a compute platform, playing a crucial role in Wing's vision for cloud-oriented programming. As such, it's vital to enable smooth communication between these functions, and what better way to do that than with JSON!
+
+JSON has become the de facto standard for data interchange on the web. As a minimal, text-based format that is a subset of JavaScript, JSON is easy to read and focuses on solving a single problem: data interchange. Its simplicity and widespread adoption make JSON a clear choice for incorporating into Wing, helping us achieve the vision of a truly cloud-oriented programming language.
+
+## Introducing Wing's Json
+
+Wing has decided to treat JSON as a first-class citizen with built-in primitives. Let's take a look at a simple example:
+
+```js
+let j = {
+  wow: "so easy"
+};
+
+// Optionally we can also add the Json keyword prefix
+let j = Json {
+  wow: "so easy"
+};
+```
+
+That's all it takes to create a JSON object in Wing. No need to import anything!
+
+Now, let's create an inflight function that requires a Json object as input:
+
+```js
+bring cloud;
+
+let b = new cloud.Bucket();
+
+let someService = new cloud.Function(inflight (jsonString: str) => {
+  let message = Json.parse(jsonString);
+  let id = message.get("id");
+  let content = str.fromJson(message.get("content"));
+  b.put("file-${id}.txt", content);
+});
+
+// Test it out
+test "storing a message" {
+  let payload = Json {
+    id: "abc123",
+    content: "Hello, Json!"
+  };
+  someService.invoke(Json.stringify(payload));
+
+  // retrieve object from bucket and validate content
+  let storedMessage = b.get("file-${str.fromJson(payload.get("id"))}.txt");
+  assert(storedMessage == str.fromJson(payload.get("content")));
+}
+```
+
+Something noteworthy happens above when we call `str.fromJson()`. All fields within a `Json` object are of type `Json`, and Wing does not allow for implicit type conversions from `Json` to other types. Thus, we must explicitly convert the `Json` object to a string using the `fromJson()` function, which exists for all Wing primitive types.
+
+Let's dive a little deeper and see what else Wing's `Json` type offers.
+
+### Converting To Structs
+
+While `Json` objects are great for interoperability, they are not the best for working with data in Wing. This is because `Json` objects are not strongly typed. However, Wing does provide a way to convert `Json` objects to structs. This conversion is done by using the `fromJson()` function, which exists on all structs created in Wing.
```

Comment:
Shouldn't we also mark struct as a code-word (`struct`)? Feel free to keep it as is, of course.
```suggestion
While `Json` objects are great for interoperability, they are not the best for working with data in Wing. This is because `Json` objects are not strongly typed. However, Wing does provide a way to convert `Json` objects to `struct` instances. This conversion is done by using the static `fromJson()` method, which exists on all `struct`s created in Wing.
```

Reply:
I don't think it's needed. A structs is a concept in wing, like classes, loops, modules, etc.

The keyword `struct` in the language is a way to declare structs, not the other way around.